### ISSUE TITLE
Color not supported with multi-entry stacked histplot

### ISSUE THREAD

**Issue by douglasdavis** [scikit-hep/mplhep#53]
If `color=` is passed to `histplot` we get a single color for all histograms (the first entry in the list). Since it's a keyword argument just passed off to matplotlib in a loop here here: https://github.com/scikit-hep/mplhep/blob/8481869bd4c64ba44301e28f9a99c5bd735bd316/mplhep/plot.py#L173-L179
It looks like color needs to be addressed in the loop so we use `color[i]`. Since the whole list of colors is being fed to Matplotlib on each iteration of the loop I guess it just takes the first entry. I'll try to come up with a patch when I have some time, but for now I just wanted to share.

minimal reproducer:
```python
import matplotlib as mpl
mpl.use("Agg")
import matplotlib.pyplot as plt
import numpy as np
from mplhep import histplot

bins = np.linspace(-3, 3, 20)
x = np.random.randn(10000)
h1, _ = np.histogram(x, bins=bins)
h2, _ = np.histogram(x, bins=bins, weights=np.ones_like(x) * 0.50)
h3, _ = np.histogram(x, bins=bins, weights=np.ones_like(x) * 0.25)

fig, ax = plt.subplots()
histplot(
    h=[h1, h2, h3],
    bins=bins,
    label=["1", "2", "3"],
    stack=True,
    histtype="fill",
    color=["C2", "C5", "C0"],
    ax=ax,
)

fig.savefig("test.png")
```

![test](https://user-images.githubusercontent.com/3202090/74599267-1b5cec80-504d-11ea-8680-5f8a8e572e85.png)

If the color argument is removed we get auto coloring

![test](https://user-images.githubusercontent.com/3202090/74599277-3a5b7e80-504d-11ea-9d49-42c639c9706a.png)

**Comment by douglasdavis** [comment]
For reference; here's a skeleton of an alternative method that works nicely:

```python
def bin_centers(edges):
    return (edges[1:] + edges[:-1]) * 0.5


def stackhist(h, edges, ax=None, **kwargs):
    if ax is None:
        ax = plt.gca()
    centers = bin_centers(edges)
    return ax.hist(
        [centers for _ in range(len(h))],
        bins=edges,
        weights=h,
        histtype="stepfilled",
        stacked=True,
        **kwargs
    )


bins = np.linspace(-3, 3, 20)
x = np.random.randn(10000)
h1, _ = np.histogram(x, bins=bins)
h2, _ = np.histogram(x, bins=bins, weights=np.ones_like(x) * 0.50)
h3, _ = np.histogram(x, bins=bins, weights=np.ones_like(x) * 0.25)

fig, ax = plt.subplots()
stackhist([h1, h2, h3], bins, ax=ax, color=["C2", "C5", "C0"], label=["1", "2", "3"])

fig.savefig("test.png")
```

**Comment by andrzejnovak** [comment]
Yup, that's an oversight. PR is welcome :)

**Comment by andrzejnovak** [comment]
@douglasdavis  Ok, I realized this actually needed to be generalized for any kwargs, so I did the PR myself. #54 Should fix this.

**Comment by andrzejnovak** [comment]
Fixed in 0.0.37
